File: 프로그래머스/0/181894. 2의 영역/2의 영역.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr) {
        int[] answer = {};
        
        List<Integer> list = new ArrayList<>();
        
        for(int i = 0; i < arr.length; i++){
            if(arr[i] == 2) list.add(i);
        }

        if(list.size() > 1){
            answer = Arrays.copyOfRange(arr,list.get(0),list.get(list.size()-1)+1);
        }else{
            answer = new int[1];
            answer[0] = list.isEmpty() == true ? -1 : arr[list.get(0)];
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181894. 2의 영역/2의 영역.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int arr[], size_t arr_len) {
    int start = 0;
    int end = arr_len - 1;
    int idx = 0;
    int len = 0;
    
    for(; start < arr_len; start++){
        if(arr[start] == 2) break;
    }
    for(; end >= 0; end--){
        if(arr[end] == 2) break;
    }
    
    if(start > end) {
        len = 1;
    }else {
        len = end - start + 1;
    }

    int* answer = (int*)malloc(sizeof(int) * len);
    
    if(end < 0){
        answer[0] = -1;
    }else{
        for(; start <= end; start++){
            answer[idx++] = arr[start];
        }
    }
    
    return answer;
}